app/models/replication_transfer.rb

```diff
@@ -21,10 +21,10 @@ def self.find_fields
 
 
 
-  def cancel!(reason)
+  def cancel!(reason, detail)
     unless cancelled
       transaction do
-        update!(cancelled: true, cancel_reason: reason)
+        update!(cancelled: true, cancel_reason: reason, cancel_reason_detail: detail)
         bag_man_request&.cancel!(reason)
       end
     end
```

Comment:
OK, let's spin off an issue for the backlog, as this point is bigger than this PR alone.